client/sites/settings/controller.tsx

```diff
@@ -104,7 +104,7 @@ export function redirectToSiteSettingsIfAdvancedHostingFeaturesNotSupported(
 	const state = context.store.getState();
 	const site = getSelectedSite( state );
 
-	if ( areAdvancedHostingFeaturesSupported( state ) === false ) {
+	if ( ! areAdvancedHostingFeaturesSupported( state ) ) {
```

Comment:
Weirdly I'm getting redirected for an atomic site too.

/sites/settings/database/deantestwoadev.wpcomstaging.com → /sites/settings/site/deantestwoadev.wpcomstaging.com

Reply:
I've spent some time having a look. The issue is that the features are never loaded when the middleware gets executed and therefore navigates to the general settings page. I've tried to get the features to load but couldn't find  a best practice way. I'm going to restore my change and file a bug. The bug is that on Simple sites users can navigate to the specific advanced features pages but won't be able to use them.

Reply:
Created an issue for it here:
- https://github.com/Automattic/wp-calypso/issues/102320